monai/data/dataset.py

```diff
@@ -77,15 +69,17 @@ class Dataset(_TorchDataset):
          },                           },                           }]
     """
 
-    def __init__(self, data: Sequence, transform: Callable | None = None) -> None:
+    def __init__(self, data: Sequence, transform: Sequence[Callable] | Callable | None = None) -> None:
         """
         Args:
             data: input data to load and transform to generate dataset for model.
             transform: a callable data transform on input data.
-
         """
         self.data = data
-        self.transform: Any = transform
+        try:
+            self.transform = Compose(transform) if not isinstance(transform, Compose) else transform
```

Comment:
Looks we miss support for transform to be None here.

Reply:
I've intentionally left this out as the `Compose` should handle the case where `transform` happens to be `None`. 

See:
https://github.com/Project-MONAI/MONAI/blob/e5afa4341b18149ca8d44f680f669b33ab423722/monai/transforms/compose.py#L238

https://github.com/Project-MONAI/MONAI/blob/e5afa4341b18149ca8d44f680f669b33ab423722/monai/transforms/compose.py#L105

Reply:
Still suggest add a check in the dataset, it's not easy to understand even it's None, it will still return a compose object. What do you think?

Reply:
Is there any reason we need to have an explicit `None` check here?

Since `Compose` will handle the case when `transform = None` internally by assigning `self.transforms = []` which will give the exact same output as before (returning the data as is), I do not see the need to add an explicit `None` check. 

Moreover, the `Compose` logic is now consistent across dataset implementations making the behaviour much more understandable in my opinion.

Reply:
Yes, I know the point, just feel it's not straightforward when users check ds.transform, it will still return a compose object.
```
import torch
from monai.data import Dataset

data = [torch.rand(10, 10) for _ in range(100)]
ds = Dataset(data=data, transform=None)
print(ds.transform) # --> <monai.transforms.compose.Compose object at 0x7ffa6088b250>
print(ds.transform.transforms) # --> ()
```
But it's ok if other developers also agree on this.
cc @ericspod @atbenmurray @Nic-Ma

Reply:
I think this behaviour without a check is fine so long as the docstring line documenting `transform` states this, currently it makes no mention of what `None` would imply or what a sequence of callables would imply.

Reply:
@ericspod @KumoLiu Docstring for this is updated in the latest commit.